org.opent2t.sample.thermostat.superpopular/com.wink.thermostat/js/thingTranslator.js

```diff
@@ -279,6 +279,149 @@ function validateResourceGet(resourceId) {
     }
 }
 
+/**
+ * If the user provides a unit, validate the value is within the units range.
+ *  Range is defined by min/max_min/max_set_point if available, otherwise:
+ *    c [9-32]
+ *    f [50-90]
+ * 
+ * If the unit is provided and within range, returns provided unit 
+ *  otherwise an exception is thrown.
+ * 
+ * If no unit is provided and the value is within one of the two ranges,
+ *  the unit of the valid range will be returned, otherwise an 
+ *  exception is thrown.
+ *  * 
+ * @param {*} resourceSchema 
+ * @param {*} providerSchema 
+ */
+function getValidatedUnit(resourceSchema, stateReader) {
+    var value = resourceSchema.temperature;
+    if (resourceSchema.hasOwnProperty('units') && resourceSchema.units != null) {
```

Comment:
same observation here. "units" may be undefined, at least theoretically.

Reply:
Fixed.